packages/stream_video/lib/src/call/session/call_session.dart

```diff
@@ -399,61 +404,65 @@ class CallSession extends Disposable {
 
-      await _eventsSubscription?.cancel();
-      _eventsSubscription = sfuWS.events.listen(_onSfuEvent);
-      await sfuWS.connect();
-
-      sfuWS.send(
-        sfu_events.SfuRequest(
-          joinRequest: sfu_events.JoinRequest(
-            clientDetails: _clientDetails,
-            token: config.sfuToken,
-            sessionId: sessionId,
-            subscriberSdp: subscriberSdp,
-            publisherSdp: publisherSdp,
-            reconnectDetails:
-                await getReconnectDetails(SfuReconnectionStrategy.fast),
-            preferredPublishOptions:
-                rtcManager?.publishOptions.map((o) => o.toDTO()),
+      Result<({SfuCallState callState, Duration fastReconnectDeadline})?>?
+          result;
+
+      if (!sfuWS.isConnected) {
+        _logger.d(() => '[fastReconnect] sfu not connected, recreating');
+        await sfuWS.recreate();
+
+        _logger.d(() => '[fastReconnect] sfu connected, sending join request');
+        sfuWS.send(
+          sfu_events.SfuRequest(
+            joinRequest: sfu_events.JoinRequest(
+              clientDetails: _clientDetails,
+              token: config.sfuToken,
+              sessionId: sessionId,
+              subscriberSdp: subscriberSdp,
+              publisherSdp: publisherSdp,
+              reconnectDetails:
+                  await getReconnectDetails(SfuReconnectionStrategy.fast),
+              preferredPublishOptions:
+                  rtcManager?.publishOptions.map((o) => o.toDTO()),
+            ),
           ),
-        ),
-      );
-
-      _logger.v(() => '[fastReconnect] wait for SfuJoinResponseEvent');
-      final event = await sfuWS.events.waitFor<SfuJoinResponseEvent>(
-        timeLimit: const Duration(seconds: 30),
-      );
+        );
 
-      if (event.isReconnected) {
-        _logger.v(
-          () =>
-              '[fastReconnect] fast-reconnect possible - requesting ICE restarts',
+        _logger.v(() => '[fastReconnect] wait for SfuJoinResponseEvent');
+        final event = await sfuWS.events.waitFor<SfuJoinResponseEvent>(
+          timeLimit: const Duration(seconds: 30),
         );
 
-        await rtcManager?.publisher?.pc.restartIce();
+        if (event.isReconnected) {
+          _logger.v(() => '[fastReconnect] fast-reconnect done');
 
-        final remoteTracks =
-            rtcManager!.tracks.values.whereType<RtcRemoteTrack>().toList();
+          stateManager.sfuPinsUpdated(event.callState.pins);
 
-        for (final track in remoteTracks) {
-          await _onRemoteTrackReceived(rtcManager!.subscriber, track);
+          result = Result.success(
+            (
+              callState: event.callState,
+              fastReconnectDeadline: event.fastReconnectDeadline
+            ),
+          );
+        } else {
+          _logger.v(() => '[fastReconnect] fast-reconnect not possible');
+          return const Result.failure(
+            VideoError(message: 'Fast reconnect not possible'),
+          );
```

Comment:
At some point we need to improve our exceptions so we can show better information to the users.